modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -473,92 +582,193 @@ else if (!json.containsKey(tag) || json.get(tag)==null){
+     * {@link LineJoin#MITER} to an expression that evaluates to "mitre".</p>
+     * 
+     * @param json The json object containing the property
+     * @param tag The json key corresponding to the property
+     * @param enumeration The Mapbox enumeration that the value should be an instance of
+     * @param fallback The fallback enumeration value, if the value is missing or invalid for the provided enumeration.
+     * @return A GeoTools expression corresponding to the Mapbox enumeration value, evaluating to a GeoTools constant.
+     */
+    public <T extends Enum<?>> Expression enumToExpression(JSONObject json, String tag,
+            Class<T> enumeration, T fallback) {
+        // Function name is inconsistent because "enum" is not a valid function name.
+        Object value = json.get(tag);
+        if (value == null) {
+            return ff.literal(convertEnumValueToGeoToolsConstant(fallback));
+        } else if (value instanceof String) {
+            // step 1 look up enumValue
+            String stringVal = (String) value;
+            if ("".equals(stringVal.trim())) {
+                return ff.literal(convertEnumValueToGeoToolsConstant(fallback));
+            }
+            T enumValue = null;
+            for (T constant : enumeration.getEnumConstants()) {
+                if (constant.toString().equalsIgnoreCase(stringVal.trim())) {
+                    enumValue = constant;
+                    break;
+                }
+            }
+            if (enumValue == null) {
+                LOGGER.warning("\"" + stringVal + "\" invalid value for enumeration "
+                        + enumeration.getSimpleName() + ", falling back to default value.");
+                return ff.literal(convertEnumValueToGeoToolsConstant(fallback));
+            }
+            // step 2 - convert to geotools constant
+            // (for now just convert to lower case)
+            //
+            return ff.literal(convertEnumValueToGeoToolsConstant(enumValue));
+        } else if (value instanceof JSONObject) {
+            MBFunction function = new MBFunction(this, (JSONObject) value);
+            return function.enumeration(enumeration);
+        } else {
+            throw new MBFormatException(
+                    "Conversion of \"" + tag + "\" value from " + value.getClass().getSimpleName()
+                            + " to " + enumeration.getSimpleName() + " not supported.");
+        }
+    }
+
+    /**
+     * Transform a Mapbox enumeration value to the corresponding GeoTools constant. For example, converts {@link LineJoin#BEVEL} to the
+     * String value "bevel", or {@link LineJoin#MITER} to the String value "mitre".
+     * 
+     * @param enumValue The Mapbox enumeration value.
+     * @return The GeoTools constant.
+     */
+    private Object convertEnumValueToGeoToolsConstant(Enum<?> enumValue) {
+
+        if (enumValue instanceof LineJoin && LineJoin.MITER.equals(enumValue)) {
+            return ff.literal("mitre");
```

Comment:
This magic needs to move to MBFunction.constant( object, Enumernation) method so we have a single location for such things. Or we could remove MBFunction.constant and just call this method instead.